src/azul/indexer/index_controller.py

```diff
@@ -158,7 +171,12 @@ def _validate_notification(self, notification):
     def contribute(self, event: Iterable[SQSRecord], *, retry=False):
         for record in event:
             message = json.loads(record.body)
-            attempts = record.to_dict()['attributes']['ApproximateReceiveCount']
+            attempts = message['attempts']
+            if attempts > 1 and not retry:
+                log.warning(f"Non-retry worker skipping message {message}.")
+                raise ContributeLambdaError('Worker already attempted digesting message.')
+            receive_count = record.to_dict()['attributes']['ApproximateReceiveCount']
```

Comment:
Another nitpick, but personally I'd covert to `int` on the same line so that if we end reusing the variable we won't have to repeat the conversion.
```suggestion
            receive_count = int(record.to_dict()['attributes']['ApproximateReceiveCount'])
```